### ISSUE TITLE
Guard  against order with `html-webpack-plugin` always triggers

### ISSUE THREAD

**Issue by posva** [Runjuu/html-inline-css-webpack-plugin#15]
Hi,

It seems that the error about ordering this plugin after HTMLWebpackPlugin always triggers no matter the order:

```js
const HtmlWebpackPlugin = require('html-webpack-plugin')
const HTMLInlineCSSWebpackPlugin = require('html-inline-css-webpack-plugin')
  .default
const path = require('path')

module.exports = () => {
  /** @type {import('webpack').Configuration} */
  const config = {
    entry: path.resolve(__dirname, './index.js'),
    mode: 'production',

    plugins: [
      new HtmlWebpackPlugin(),
      new HTMLInlineCSSWebpackPlugin(),
    ],
  }

  return config
}
```

**Comment by Runjuu** [comment]
Hi @posva , can you provide a minimal demo that can produce the issue?

**Comment by posva** [comment]
Sure, I realized it happens with the beta version of html webpack plugin: https://github.com/posva/webpack-css-inline-report

**Comment by Runjuu** [comment]
Seems the beta version of `html-webpack-plugin` are using a new way to executes tapable hooks. I will fix the issue ASAP.

**Comment by posva** [comment]
I'm glad you find the culprit!
No rushes, whenever you find the time :)

**Comment by Runjuu** [comment]
Hi @posva , I fixed the issue at `1.8.0`. Let me know if there is any problem

**Comment by posva** [comment]
Seems to be working perfectly
